test/test_other.py

```diff
@@ -13996,3 +13996,22 @@ def test_noExitRuntime_deps(self):
 ''')
     err = self.expect_fail([EMCC, test_file('hello_world.c'), '--js-library=lib.js', '-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE=foo'])
     self.assertContained('error: noExitRuntime cannot be referenced via __deps mechansim', err)
+
+  @also_with_wasm64
+  def test_return_emval(self):
+    create_file('test.cpp', r'''
+    #include <emscripten.h>
+    #include <emscripten/val.h>
+
+    using namespace emscripten;
+
+    EM_JS(EM_VAL, get_val, (), {
+      return Emval.toHandle(0);
+    });
+
+    int main() {
+      return val::take_ownership(get_val()).as<int>();
+    }
+    ''')
+    self.emcc_args += ['--bind']
+    self.do_runf('test.cpp')
```

Comment:
We run the entire core test suite in wasm64 mode.. do none of those tests use `toHandle`?

Reply:
I see test_embind_val (`test/embind/test_val.cpp`)  calls `Emval.toHandle()`, but I guess it does it from EM_ASM instead of EM_JS.

I would hope that test could be extended to cover this case perhaps?

Reply:
> but I guess it does it from EM_ASM instead of EM_JS.

Hm yeah I see it now... but why does that one pass? Because of EM_ASM's different encoding of params/results?

Reply:
I wish I knew. I thought you know EM_ASM internals more than me, I just know that it's complicated and that it's somehow different from EM_JS 😅

Reply:
Ah I think that test works because it uses explicit `EM_ASM_PTR` which expects a pointer and so is correct size on any platform.

Reply:
In fact, it breaks after the change to `enum class: uint32_t` I just made...

Reply:
I think that breaking is reasonable.. EM_ASM_INT should do it.